Cython/Compiler/Parsing.py

```diff
@@ -2340,7 +2342,25 @@ def p_statement_list(s, ctx, first_statement = 0):
     pos = s.position()
     stats = []
     while s.sy not in ('DEDENT', 'EOF'):
-        stat = p_statement(s, ctx, first_statement = first_statement)
+        # When processing statements in a fault-tolerant mode, it's possible
+        # that because some Node isn't properly constructed, internal errors
+        # happen (for instance if the right hand side of an assignment would
+        # be `None`, trying to access its `pos` would throw an
+        # AttributeError). For the fault-tolerant mode this is ok (we don't
+        # want to change the whole parser to accommodate that use-case, so,
+        # just report that as a `CompileError` so that it can be seen and
+        # proceed as usual).
+        stat = None
+        try:
+            stat = p_statement(s, ctx, first_statement = first_statement)
+        except Exception as exc:
+            if s.fault_tolerant:
+                if not isinstance(exc, CompileError):
+                    exc = CompileError(pos, "Internal Error: "+ str(exc))
+                report_error(exc)
+                s.next()
```

Comment:
Put a `continue` here so that the node is not added (that way you also don`t need the `stat=None` assignment in the start of this function.

Reply:
I notice that using `continue` without assigning stat a value causes errors ahead on the code. If it continue the loop without changing stat var, it will store wrong values at the AST

example: if the statement before the errored statement is a PassStatementNode, the errored statement will also be stored as a PassStatementNode

Reply:
The code ahead needs the stat value, even if it is None, because the DefStatement is caught and it need to have a body type. What is raising errors to get identified is the statement inside the def, not the DefStatement itself